### PR TITLE
Add feedback to API

### PR DESCRIPTION
**Pull request by samhinshaw**
Add the `feedback()` command to the nbgrader API, for programmatic feedback generation in Python.

### CODE REVIEW

**Comment by samhinshaw** [comment]
The API call seems to work correctly, but I had some trouble writing a proper test. Relevant error also occurs when testing locally: https://travis-ci.org/jupyter/nbgrader/jobs/416929082#L1363

**Comment by jhamrick** [comment]
Thanks! From a cursory glance this looks great; I'll have a closer look over the next few days.

File: nbgrader/apps/api.py [hunk 1]
```diff
@@ -861,6 +861,36 @@ def assign(self, assignment_id, force=True, create=True):
...
+
+            - success (bool): whether or not the operation completed successfully
+            - error (string): formatted traceback
+            - log (string): captured log output
+
+        """
+        with temp_attrs(self.coursedir, assignment_id=assignment_id):
+            app = Feedback(coursedir=self.coursedir, parent=self)
+            app.force = force
+            app.create_assignment = create
```
Comment (jhamrick) [inline nbgrader/apps/api.py]: I think `create_assignment` is only relevant for `nbgrader assign` and won't do anything for `nbgrader feedback`, so you should be able to remove this line.

File: nbgrader/apps/api.py [hunk 2]
```diff
@@ -861,6 +861,36 @@ def assign(self, assignment_id, force=True, create=True):
             app.create_assignment = create
             return capture_log(app)
 
+    def feedback(self, assignment_id, force=True, create=True):
```
Comment (jhamrick) [inline nbgrader/apps/api.py]: I think this function should also take a student_id (as with the `autograde` method) and only generate feedback for one student at a time.

**Comment by jhamrick** [comment]
Hi @samhinshaw just wanted to see if you ever had a chance to look more at this? I think we may be looking at generating feedback from the formgrader during the Edinburgh hackathon over the next few days (#1079) and this would be a prerequisite for that so it would be great to get this merged. If you haven't had time to work on it further, that's totally fine---we can use what you've done as a starting point and finish it up for you if you'd like 🙂

**Comment by ttimbers** [comment]
@jhamrick - @samhinshaw is no longer actively working with me on the project this PR was related to. However, we would still love this functionality, we've just had no one able to work on this. So, if this can be worked on at the Edinburgh hackathon that would be amazing!

**Comment by jhamrick** [comment]
Thanks for the update @ttimbers! It sounds like something a few other people want as well, so hopefully it will get checked off the list 🙂

**Comment by jhamrick** [comment]
This is going to be superseded by #1092 so I am going to go ahead and close this. Thanks for the work that you put into this originally @samhinshaw !

**Pull request closed without merging**